### ISSUE TITLE
CTRL+Cing console apps started dotnet run doesn't wait for the target to terminate

### ISSUE THREAD

**Issue by Trolldemorted** [dotnet/cli#11050]
# Original issue (https://github.com/dotnet/core/issues/2530):

Let's consider this brief demo:

```c#
private static readonly CancellationTokenSource CancelSource = new CancellationTokenSource();
public static void Main()
{
    Console.WriteLine("Hello world!");
    Console.CancelKeyPress += (s, e) =>
    {
        e.Cancel = true;
        CancelSource.Cancel();
        Console.WriteLine("CancelKeyPress event handler finished");
    };
    CancelSource.Token.WaitHandle.WaitOne();
    Task.Delay(1000).Wait();
    Console.WriteLine("Main finished");
}
```
As you can see, the Program is waiting for the cancel event, stating that it does not want to be shut down, waiting one second, printing "Main finished" and exiting.

I would expect the output to be:
```
C:\Users\Benni\source\repos\Test\Test>dotnet run
Hello world!
CancelKeyPress event handler finished
^C
Main finished
C:\Users\Benni\source\repos\Test\Test>
```
But it is:
```
C:\Users\Benni\source\repos\Test\Test>dotnet run
Hello world!
CancelKeyPress event handler finished
^C
C:\Users\Benni\source\repos\Test\Test>Main finished
```
Unfortunately the shells (I tested bash, cmd.exe and powershell) print the CWD (for the next command) before the program has actually finished running!

# .NET CLI's role

We managed to narrow the problem down to the usage of `dotnet run`. If you run the produced dll directly (e.g. `dotnet .\bin\Debug\netcoreapp2.2\Test.dll` the program works as expected, if you use `dotnet run` it does not.

I will be closing my original issue, and hope here is the right place to report this.

## Affected systems

When building with dotnet sdk 2.2.105 the problem is reproducible on W10 1809 and on Archlinux. A [stackoverflow user states](https://stackoverflow.com/questions/55441329/cancelkeypress-handler-confuses-shells-cmd-exe-powershell-and-bash) that on his system the shells behave differently and the problem does not occur, but he or she is using dotnet sdk v2.2.103.

**Comment by Trolldemorted** [comment]
I just downloaded and installed 3.0.100, it is indeed running fine.

**Comment by peterhuene** [comment]
Thanks for confirming @Trolldemorted!  Unfortunately the fix is not something that will meet the servicing bar to backport to a 2.1/2.2 .NET Core SDK.  As such, I'm going to close this issue as fixed in the future release of 3.0.  Thank you!
